### ISSUE TITLE
GC not working hard enough on bigarrays

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7495]
**Original bug ID:** 7495
**Reporter:** @mmottl
**Assigned to:** @damiendoligez
**Status:** acknowledged (set by @mshinwell on 2017-03-07T12:33:34Z)
**Resolution:** open
**Priority:** normal
**Severity:** minor
**Version:** 4.04.0
**Category:** runtime system and C interface
**Monitored by:** @mshinwell @mmottl

## Bug description

Consider the following code:

```ocaml
let () =
  let open Bigarray in
  for _i = 1 to 100_000 do
    let vec = Array1.create float64 c_layout 1_000_000 in
    (* Makes sure that physical memory is actually allocated *)
    Array1.fill vec 0.
  done
```

This code will let memory consumption quickly go into the GB range.  A comparable implementation using float arrays will stay stable in the low MB range:

```ocaml
let () =
  for _i = 1 to 100_000 do
    ignore (Array.make 1_000_000 0.)
  done
```

I remember that this is a long-standing problem, probably the same as #4616, and likely affects custom block allocation in general.  Not sure whether it's already tracked in another still active issue.  Since the GC will likely undergo significant changes in the near future due to attempts to parallelize it, I think this issue should remain on people's mind.  Some practical numerical code can run into problems with this limitation.

**Comment by vicuna** [comment]
**Comment author:** @dra27

Is this the same issue as #4616, as suggested (which was yours)? It looks so to me...

**Comment by xavierleroy** [comment]
Cannot reproduce with 4.09 (didn't check anything older).  Was probably fixed as part of #1738.

**Comment by mmottl** [comment]
Yes, I'd expect that #1738 fixed this.
